File: Backend/src/main/java/com/frs/timetracker/domain/TimeEntry.java
package com.frs.timetracker.domain;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.AllArgsConstructor;
import lombok.Data;

import javax.persistence.*;
import java.time.Instant;
import java.util.Date;

@Data
@Entity
@AllArgsConstructor
public class TimeEntry {
    @Id
    @GeneratedValue
    private long id;
    private long beginn;
    private long end;
    @JsonBackReference(value = "company")
    @ManyToOne
    private Company company;
    private Date createdAt;
    @JsonBackReference(value = "user")
    @ManyToOne
    private User user;
    private long timeWorked;

    public TimeEntry(){
    }

    public TimeEntry(long beginn, long end, Company company){
        this.beginn = beginn;
        this.end = end;
        this.company = company;
    }
    @PrePersist
    void createdAt(){this.createdAt= new Date();}

    public void setTimeWorked(){
        this.timeWorked = this.end - this.beginn;
    }


}

File: Frontend/src/app/domain/TimeEntry.ts
import {Moment} from 'moment';
import {User} from './User';
import {Company} from './Company';

export interface ITimeEntry {
  id?: number;
  beginn?: number;
  end?: number;
  user?: User;
  company?: Company;
  createdAt?: Moment;
  timeWorked?: number;
}

export class TimeEntry implements ITimeEntry {
  constructor(
    public id?: number,
    public beginn?: number,
    public end?: number,
    public user?: User,
    public company?: Company,
    public createdAt?: Moment,
    public timeWorked?: number

) {

  }
}
